java/arcs/android/devtools/DevToolsService.kt

```diff
@@ -213,6 +226,24 @@ open class DevToolsService : Service() {
     devToolsServer.send(rawMessage.toJson())
   }
 
+  fun createAndSendProxyMessageProto(
+    proxyMessage: ByteArray,
+    storeType: DevtoolsMessage.DevToolsProxyMessageProto.StoreType,
+    storageKey: String
+  ) {
+    val messageProto = DevtoolsMessage.DevToolsProxyMessageProto.newBuilder()
+      .setProxyMessage(proxyMessage.decodeProxyMessage().toProto())
+      .setStoreType(storeType)
+      .setStorageKey(storageKey)
+      .build()
+
+    val proto = DevtoolsMessage.DevToolsMessageProto.newBuilder()
+      .setProxyMessage(messageProto)
+      .build()
+
+    devToolsServer.send(proto.toString())
```

Comment:
Just a thought, you don't have to do this right now, and it might not be the right call at all, but you could serialise to bytes and send them over, instead of using the toString representation. You'll be able to decode those in JS easily. It'll be faster, and possibly less error-prone (I'm not sure whether the string representation in JS will be exactly the same, though it might be)

Reply:
@piotrswigon to comment.